token/core/common/interop/htlc/validator_htlc.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package htlc
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/hyperledger-labs/fabric-token-sdk/token/core/common"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/identity"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/interop/htlc"
+	"github.com/pkg/errors"
+)
+
+// TransferHTLCValidate checks the validity of the HTLC scripts, if any
+func TransferHTLCValidate[P driver.PublicParameters, T driver.Output, TA driver.TransferAction, IA driver.IssueAction, DS driver.Deserializer](ctx *common.Context[P, T, TA, IA, DS]) error {
+	now := time.Now()
+
+	for i, in := range ctx.InputTokens {
+		owner, err := identity.UnmarshalTypedIdentity(in.GetOwner())
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal owner of input token")
+		}
+		// is it owned by an htlc script?
+		if owner.Type == htlc.ScriptType {
+			// Then, the first output must be compatible with this input.
+			if len(ctx.TransferAction.GetOutputs()) != 1 {
+				return errors.New("invalid transfer action: an htlc script only transfers the ownership of a token")
```

Comment:
Maybe we should also enforce that `len(Inputs) == 1` since a `claim` and a `reclaim` are just ownership transfers.